File: scm-java/src/main/scheme/primitives/PrimitivePsInfo.java
package scheme.primitives;
import scheme.*;

public class PrimitivePsInfo extends Primitive {
    @Override
    public String name() { return "ps-info"; }

    @Override
    public String info() {
        return "Syntax: (ps-info pid)\n" +
               "Library: (scm system)\n" +
               "Description: Returns an alist describing the process with the given pid,\n" +
               "  or #f if no such process exists or it cannot be inspected. See (ps)\n" +
               "  for the field set.\n" +
               "Example:\n" +
               "  (cdr (assq 'command (ps-info (current-pid)))) => \"scm\"";
    }

    @Override
    public Object apply(SourcePos pos, Object[] arguments) {
        checkArgs(pos, arguments, 1, 1);
        long pid = IntegerMath.toLong(arguments[0]);
        var h = ProcessHandle.of(pid);
        if (!h.isPresent()) return Value.F;
        return PrimitivePs.buildInfo(h.get());
    }
}

File: scm-csharp/scheme/primitives/PrimitivePsInfo.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Runtime.InteropServices;

namespace scheme;

public class PrimitivePsInfo : Primitive
{
    public override string Name() => "ps-info";

    public override string Info() =>
        "Syntax: (ps-info pid)\n" +
        "Library: (scm system)\n" +
        "Description: Returns an alist describing the process with the given pid, or #f\n" +
        "  if no such process exists or it cannot be inspected. See (ps) for the\n" +
        "  field set.\n" +
        "Example:\n" +
        "  (cdr (assq 'command (ps-info (current-pid)))) => \"scm\"";

    public override object Apply(SourcePos? pos, object[] arguments)
    {
        CheckArgs(pos, arguments, 1, 1);
        long pid = IntegerMath.ToLong(arguments[0]);
        Process? p;
        try { p = Process.GetProcessById((int) pid); }
        catch (Exception) { return Value.F; }
        if (p == null) return Value.F;
        bool linux = RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
        Dictionary<int, int>? ppidMap =
            RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? ProcUtil.WindowsPpidMap() : null;
        try
        {
            return PsPrimitiveHelpers.BuildInfo(p, linux, ppidMap);
        }
        finally { p.Dispose(); }
    }
}
